src/shape.rs

```diff
@@ -0,0 +1,74 @@
+use std::fmt;
+use std::ops::Index;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Shape {
```

Comment:
Do we want to treat `N x 1` as different from just `N`?

Reply:
Good question. What do you think?

Reply:
I think the consensus was that `N x 1` (and `1 x N`) and `N` are different because they have different "orders". Having said that, I think we will have something which means that `N` can be used in matrix multiplication and other operations as long as it's unambiguous.

For example `N` should be able to multiply with an `N x N` matrix. Whether it's treated as `N x 1` or `1 x N` being based on the order of the multiplication.

The return order is a bit hazy, because numpy would return an order 1 matrix if we use `N`. So maybe we make it consistent with that.